From 105b75b7d5be262544958bf60ffbd437693842c3 Mon Sep 17 00:00:00 2001
From: Andrew Boie <andrew.p.boie@intel.com>
Date: Thu, 21 Mar 2013 13:05:22 -0700
Subject: Allow access to debugfs device nodes

Even though the permissions on the individual nodes were set,
trace markers couldn't be set by the framework due to the 0700
permissions on /sys/kernel/debug.

This allows for stack traces when non-root native processes
segfault.

Change-Id: Ib1c145665595774c9e0f13759711de67508c5bdd
Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>
---
 rootdir/init.trace.rc |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/rootdir/init.trace.rc b/rootdir/init.trace.rc
index 4232474..5bf5aa7 100644
--- a/rootdir/init.trace.rc
+++ b/rootdir/init.trace.rc
@@ -3,10 +3,13 @@
 on property:ro.debuggable=1
     mount debugfs debugfs /sys/kernel/debug
 
-# Allow writing to the kernel trace log.
+    # Allow processes other than root to access device nodes
+    # under /sys/kernel/debug
+    chmod 0755 /sys/kernel/debug
+    # Allow writing to the kernel trace log.
     chmod 0222 /sys/kernel/debug/tracing/trace_marker
 
-# Allow the shell group to enable (some) kernel tracing.
+    # Allow the shell group to enable (some) kernel tracing.
     chown root shell /sys/kernel/debug/tracing/trace_clock
     chown root shell /sys/kernel/debug/tracing/buffer_size_kb
     chown root shell /sys/kernel/debug/tracing/options/overwrite
@@ -29,6 +32,6 @@ on property:ro.debuggable=1
     chmod 0664 /sys/kernel/debug/tracing/events/cpufreq_interactive/enable
     chmod 0664 /sys/kernel/debug/tracing/tracing_on
 
-# Allow only the shell group to read and truncate the kernel trace.
+    # Allow only the shell group to read and truncate the kernel trace.
     chown root shell /sys/kernel/debug/tracing/trace
     chmod 0660 /sys/kernel/debug/tracing/trace
-- 
1.7.4.1

